ingestion_program/datasets.py

```diff
@@ -201,25 +201,38 @@ def Neurips2024_public_dataset():
         zipfile.BadZipFile: If the downloaded file is not a valid zip file.
     """
     current_path = Path.cwd()
-    file_read_loc = current_path / "public_data"
-    if not file_read_loc.exists():
-        file_read_loc.mkdir()
-
-    url = "https://www.codabench.org/datasets/download/9c99a23c-f199-405a-b795-b42ea2dd652d/"
-    file = file_read_loc / "public_data.zip"
-    chunk_size = 1024 * 1024
-    if not file.exists():
-        with requests.get(url, stream=True) as r:
-            r.raise_for_status()
-            with file.open("wb") as f:
-                for chunk in r.iter_content(chunk_size=chunk_size):
-                    f.write(chunk)
-
-    input_data = file_read_loc / "input_data"
-    if not input_data.exists():
-        with ZipFile(file) as zip:
-            zip.extractall(path=file_read_loc)
-
-    return Data(
-        str(current_path / "public_data" / "input_data")
-    )
+    public_data_folder_path = os.path.join(current_path, "public_data")
+    public_input_data_folder_path = os.path.join(current_path, "public_data", "input_data")
+    public_data_zip_path = os.path.join(current_path, "public_data.zip")
```

Comment:
We should probably stick with `pathlib` as that is how the rest of this code performs path manipulation, rather than mixing the two.

Reply:
I see that only `current_path` is using Pathlib, I will update it to use os.path everywhere. What do you think @cjh1

Reply:
I would prefer `pathlib` but `os.path` would work too, just having it consistent would be great.